### ISSUE TITLE
IAM Permissions

### ISSUE THREAD

**Issue by slajax** [apex/apex#93]
What are the best practices for setting up IAM permissions for apex?

**Comment by slajax** [comment]
```
- Go to IAM Management Console
- Under Groups create a group named apex
- Assign: AWSLambdaFullAccess, AmazonAPIGatewayAdministrator, AWSLambdaRole
- Under Users create a user named apex
- Add apex user to apex group
- Generate Access Key for apex user
- export as per instructions in read me
```
This seemed to work for me but I noticed your example pointed at the lambda role directly. Can the permissions be lowered to non service admin levels?

Let me know and I can put something in the wiki, new users will need this info.

**Comment by tj** [comment]
Depends on the use-case and what the function(s) should have access to. The AWS docs for Lambda go into detail there but we'll have some sort of bootstrapping/config there eventually. Terraform is better for managing that kind of thing but it would be nice if the basics at least work out of the box

**Comment by slajax** [comment]
Yeah obviously you might need to assign broader roles depending on the functionality I'm just curious what the defaults are to get going with a quick hello world that has no external dependencies. These are the ones I used to get going. I can add something about this to the wiki if you like. I've been meaning to look at terraform, so as I ramp up on that I can add update it to talk to the terraform piece.

**Comment by mthenw** [comment]
There are two different things:

- role that each function gets assigned to - and here you don't need any policy, just empty role is fine for quick hello world with `apex invoke`

- account needed for deploying (managing) Lambda funcs with `apex deploy` etc. For that account you need to have AWSLambdaFullAccess policy (through user's permissions or through group as you mentioned in your config, IMHO more straightforward is to assign this role to user).

**Comment by slajax** [comment]
Closed by [b1012a5801282f382746c69c6c15a0ca7a46344c](https://github.com/apex/apex/wiki/Getting-Started/b1012a5801282f382746c69c6c15a0ca7a46344c)
